### ISSUE TITLE
[x64]Test failed:Regressions_coreclr._16775_aredinterfacemethod_aredinterfacemethod_._16775_sharedinterfacemethod_sharedinterfacemethod_sh

### ISSUE THREAD

**Issue by AriNuer** [dotnet/coreclr#20943]
**Job:**
https://ci.dot.net/job/dotnet_coreclr/job/master/view/x64/job/jitstress/job/x64_checked_osx10.12_gcstress0x3_tst/76/

**Failed test:**
Regressions_coreclr._16775_aredinterfacemethod_aredinterfacemethod_._16775_sharedinterfacemethod_sharedinterfacemethod_sh

**Log:**
```
Regressions_coreclr._16775_aredinterfacemethod_aredinterfacemethod_._16775_sharedinterfacemethod_sharedinterfacemethod_sh [FAIL]
        /Users/dotnet-bot/j/workspace/dotnet_coreclr/master/jitstress/x64_checked_osx10.12_gcstress0x3_tst/bin/tests/OSX.x64.Checked/Regressions/coreclr/16775/sharedinterfacemethod/sharedinterfacemethod.sh: line 244: 47294 Segmentation fault: 11  $_DebuggerFullPath "$CORE_ROOT/corerun" $ExePath $CLRTestExecutionArguments
        /n/nReturn code:      1/nRaw output file:      /Users/dotnet-bot/j/workspace/dotnet_coreclr/master/jitstress/x64_checked_osx10.12_gcstress0x3_tst/bin/tests/OSX.x64.Checked/Regressions/coreclr/Reports/Regressions.coreclr/16775/sharedinterfacemethod/sharedinterfacemethod.output.txt/nRaw output:/nBEGIN EXECUTION
        /Users/dotnet-bot/j/workspace/dotnet_coreclr/master/jitstress/x64_checked_osx10.12_gcstress0x3_tst/bin/tests/OSX.x64.Checked/Tests/Core_Root/corerun sharedinterfacemethod.exe
        Expected: 100
        Actual: 139
        END EXECUTION - FAILED
        Test Harness Exitcode is : 1
        /nTo run the test:/n> set CORE_ROOT=/Users/dotnet-bot/j/workspace/dotnet_coreclr/master/jitstress/x64_checked_osx10.12_gcstress0x3_tst/bin/tests/OSX.x64.Checked/Tests/Core_Root/n> /Users/dotnet-bot/j/workspace/dotnet_coreclr/master/jitstress/x64_checked_osx10.12_gcstress0x3_tst/bin/tests/OSX.x64.Checked/Regressions/coreclr/16775/sharedinterfacemethod/sharedinterfacemethod.sh/n
        Expected: True
        Actual:   False
        Stack Trace:
          /Users/dotnet-bot/j/workspace/dotnet_coreclr/master/checked_osx10.12/bin/tests/OSX.x64.Checked/TestWrappers/Regressions.coreclr/Regressions.coreclr.XUnitWrapper.cs(3329,0): at Regressions_coreclr._16775_aredinterfacemethod_aredinterfacemethod_._16775_sharedinterfacemethod_sharedinterfacemethod_sh()
```

**Comment by sandreenko** [comment]
Fails on windows as well, output is similar to #21044, @MichalStrehovsky

**Comment by MichalStrehovsky** [comment]
Fixed in #21525
